packages/casting/src/follower-cosmjs.js

```diff
@@ -112,290 +145,442 @@ export const makeCosmjsFollower = (
+  const provideQueryClient = async endpoint => {
     if (endpointToQueryClient.has(endpoint)) {
       // Cache hit.
       const queryClient = endpointToQueryClient.get(endpoint);
       assert(queryClient);
       return queryClient;
     }
     // Create a new client.  They retry automatically.
-    const rpcClient = await Tendermint34Client.connect(endpoint);
+    const rpcClient = await provideTendermintClient(endpoint);
     const queryClient = QueryClient.withExtensions(rpcClient);
     endpointToQueryClient.set(endpoint, queryClient);
     return queryClient;
   };
 
   /**
-   * @param {'queryVerified' | 'queryUnverified'} method
+   * @param {number} [height]
+   * @returns {Promise<Uint8Array>}
    */
-  const makeQuerier =
-    method =>
-    /**
-     * @param {number} [height]
-     * @returns {Promise<Uint8Array>}
-     */
-    async height => {
-      const {
-        storeName,
-        storeSubkey,
-        dataPrefixBytes = new Uint8Array(),
-      } = await castingSpecP;
-      assert.typeof(storeName, 'string');
-      assert(storeSubkey);
-      let queryPath;
-      switch (method) {
-        case 'queryVerified': {
-          queryPath = storeName;
-          break;
-        }
-        case 'queryUnverified': {
-          queryPath = `store/${storeName}/key`;
-          break;
-        }
-        default: {
-          assert.fail(X`unrecognized method ${method}`);
-        }
-      }
-      const values = await E(leader).mapEndpoints(where, async endpoint => {
-        const queryClient = await getOrCreateQueryClient(endpoint);
-        return E(queryClient)
-          [method](queryPath, storeSubkey, height)
-          .then(
-            result => {
-              return { result, error: null };
-            },
-            error => {
-              return { result: null, error };
-            },
-          );
+  const getProvenDataAtHeight = async height => {
+    const {
+      storeName,
+      storeSubkey,
+      dataPrefixBytes = defaultDataPrefixBytes,
+    } = await castingSpecP;
+    assert.typeof(storeName, 'string');
+    assert(storeSubkey);
+    const values = await E(leader).mapEndpoints(where, async endpoint => {
+      const queryClient = await provideQueryClient(endpoint);
+      return E(queryClient)
+        .queryVerified(storeName, storeSubkey, height)
+        .then(
+          result => {
+            return { result, error: null };
+          },
+          error => {
+            return { result: null, error };
+          },
+        );
+    });
+    const { result, error } = collectSingle(values);
+    if (error !== null) {
+      throw error;
+    }
+    assert(result);
+
+    if (result.length === 0) {
+      // No data.
+      return result;
+    }
+
+    // Handle the data prefix if any.
+    assert(
+      result.length >= dataPrefixBytes.length,
+      X`result too short for data prefix ${dataPrefixBytes}`,
+    );
+    assert(
+      dataPrefixBytes.every((v, i) => v === result[i]),
```

Comment:
```suggestion
      arrayEqual(result.subarray(0, dataPrefixBytes.length), dataPrefixBytes),
```
(we might as well use the helper consistently)

Reply:
I missed this one. Thank you!